spring-kafka/src/main/java/org/springframework/kafka/core/TransactionSuffixManager.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.core;
+
+/**
+ * The strategy for managing transactional producer suffixes.
+ *
+ * @author Ilya Starchenko
+ *
+ * @since 3.1.1
+ */
+public interface TransactionSuffixManager {
+
+	/**
+	 * Retrieve the suffix for the transactional producer.
+	 *
+	 * @param txIdPrefix the transaction id prefix.
+	 * @return the suffix.
+	 */
+	String retrieveSuffix(String txIdPrefix);
+
+	/**
+	 * Return the suffix for the transactional producer back for reuse.
+	 *
+	 * @param txIdPrefix the transaction id prefix.
+	 * @param suffix the suffix.
+	 */
+	void returnSuffix(String txIdPrefix, String suffix);
```

Comment:
`retrieve` and `return` in the API names might sound confusing. Can we change `returnSuffix` to something that is more explicit to reason about? maybe `releaseSuffixBack`? or something along those lines.

Reply:
I like those.

Reply:
Fixed.